roles/ha_apps/etcd/files/ha-etcd-template.yml

```diff
@@ -0,0 +1,94 @@
+---
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: etcd
+  labels:
+    app: etcd
+
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: etcd-client
+  namespace: etcd
+  labels:
+    app: etcd
+spec:
+  type: LoadBalancer
+  ports:
+  - name: etcd-client
+    port: 2379
+    protocol: TCP
+    targetPort: 2379
+  selector:
+    app: etcd
+
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: etcd
+  namespace: etcd
+  labels:
+    app: etcd
+spec:
+  clusterIP: None
+  ports:
+  - port: 2379
+    name: client
+  - port: 2380
+    name: peer
+  selector:
+    app: etcd
+
+---
+apiVersion: apps/v1beta1
+kind: StatefulSet
+metadata:
+  name: etcd
+  namespace: etcd
+  labels:
+    app: etcd
+spec:
+  serviceName: etcd
+  replicas: 3
+  template:
+    metadata:
+      name: etcd
+      labels:
+        app: etcd
+    spec:
+      containers:
+      - name: etcd
+        image: quay.io/coreos/etcd:latest
+        ports:
+        - containerPort: 2379
+          name: client
+        - containerPort: 2380
+          name: peer
+        volumeMounts:
+        - name: data
+          mountPath: /var/run/etcd
+        command:
+          - /bin/sh
+          - -c
+          - |
+            PEERS="etcd-0=http://etcd-0.etcd:2380,etcd-1=http://etcd-1.etcd:2380,etcd-2=http://etcd-2.etcd:2380"
+            exec etcd --name ${HOSTNAME} \
+              --listen-peer-urls http://0.0.0.0:2380 \
+              --listen-client-urls http://0.0.0.0:2379 \
+              --advertise-client-urls http://${HOSTNAME}.etcd:2379 \
+              --initial-advertise-peer-urls http://${HOSTNAME}:2380 \
+              --initial-cluster-token etcd-cluster-1 \
+              --initial-cluster ${PEERS} \
+              --initial-cluster-state new \
+              --data-dir /var/run/etcd/default.etcd
```

Comment:
You might wanna add some `liveliness` and `readiness` probes, that would make sure that it is running in `ha`. You can find an example [here](https://github.com/alaypatel07/etcd-sts-operator/blob/b7fbc581a04e337a5760c4c3f866f13655b4c2f8/roles/reconcile/templates/ss.yaml#L29). 

I would also explicitly point the etcdctl command in the probes to `LoadBalancer` service like this:
```
ETCDCTL_API=3; etcdctl --endpoints http://etcd-client.etcd:2379 endpoint status
```